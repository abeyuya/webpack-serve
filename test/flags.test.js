const { resolve } = require('path');

const { apply } = require('../lib/flags');

describe('flags', () => {
  test('apply defaults', () => {
    const argv = {};
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply devWare', () => {
    const argv = { devWare: { publicPath: '/static' } };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply hotClient', () => {
    const argv = { hotClient: { hmr: false, reload: false } };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply hotClient false', () => {
    const argv = { hotClient: false };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply hmr false', () => {
    const argv = { hmr: false };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply https cert/key', () => {
    const httpsCert = resolve(__dirname, './fixtures/test-cert.pem');
    const httpsKey = resolve(__dirname, './fixtures/test-key.pem');
    const argv = { httpsCert, httpsKey };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply https pass/pfx', () => {
    const httpsPass = 'sample';
    const httpsPfx = resolve(__dirname, './fixtures/test-cert.pfx');
    const argv = { httpsPass, httpsPfx };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply logLevel', () => {
    const argv = { logLevel: 'silent' };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply logTime', () => {
    const argv = { logTime: true };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply open', () => {
    const argv = { openApp: 'app', openPath: 'path' };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply open boolean', () => {
    const argv = { open: true };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply port string', () => {
    const argv = { port: '8888' };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply port number', () => {
    const argv = { port: 8888 };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('apply reload false', () => {
    const argv = { reload: false };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('compound logTime/logLevel', () => {
    const argv = { logLevel: 'warn', logTime: true };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('compound devMiddleware value', () => {
    const argv = { devWare: { batman: 'robin' }, logTime: true };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('compound hotClient value', () => {
    const argv = { hotClient: { hmr: false }, logTime: true };
    expect(apply(argv)).toMatchSnapshot();
  });

  test('compound hotClient: false value', () => {
    const argv = { hotClient: false, logTime: true };
    expect(apply(argv)).toMatchSnapshot();
  });
});
