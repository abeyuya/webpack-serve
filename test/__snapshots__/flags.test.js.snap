// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flags apply defaults 1`] = `Object {}`;

exports[`flags apply devWare 1`] = `
Object {
  "devMiddleware": Object {
    "publicPath": "/static",
  },
}
`;

exports[`flags apply hmr false 1`] = `
Object {
  "hotClient": Object {
    "hmr": false,
  },
}
`;

exports[`flags apply hotClient 1`] = `
Object {
  "hotClient": Object {
    "hmr": false,
    "reload": false,
  },
}
`;

exports[`flags apply hotClient false 1`] = `
Object {
  "hotClient": Object {
    "hmr": false,
  },
}
`;

exports[`flags apply https cert/key 1`] = `
Object {
  "https": Object {
    "cert": "<PROJECT_ROOT>/test/fixtures/test-cert.pem",
    "key": "<PROJECT_ROOT>/test/fixtures/test-key.pem",
  },
}
`;

exports[`flags apply https pass/pfx 1`] = `
Object {
  "https": Object {
    "passphrase": "sample",
    "pfx": "<PROJECT_ROOT>/test/fixtures/test-cert.pfx",
  },
}
`;

exports[`flags apply logLevel 1`] = `
Object {
  "devMiddleware": Object {
    "logLevel": "silent",
  },
  "hotClient": Object {
    "logLevel": "silent",
  },
  "logLevel": "silent",
}
`;

exports[`flags apply logTime 1`] = `
Object {
  "devMiddleware": Object {
    "logTime": true,
  },
  "hotClient": Object {
    "logTime": true,
  },
  "logTime": true,
}
`;

exports[`flags apply open 1`] = `
Object {
  "open": Object {
    "app": "app",
    "path": "path",
  },
}
`;

exports[`flags apply open boolean 1`] = `
Object {
  "open": Object {
    "": true,
    "path": "/",
  },
}
`;

exports[`flags apply port number 1`] = `
Object {
  "port": 8888,
}
`;

exports[`flags apply port string 1`] = `
Object {
  "port": "8888",
}
`;

exports[`flags apply reload false 1`] = `
Object {
  "hotClient": Object {
    "reload": false,
  },
}
`;

exports[`flags compound devMiddleware value 1`] = `
Object {
  "devMiddleware": Object {
    "batman": "robin",
    "logTime": true,
  },
  "hotClient": Object {
    "logTime": true,
  },
  "logTime": true,
}
`;

exports[`flags compound hotClient value 1`] = `
Object {
  "devMiddleware": Object {
    "logTime": true,
  },
  "hotClient": Object {
    "hmr": false,
    "logTime": true,
  },
  "logTime": true,
}
`;

exports[`flags compound hotClient: false value 1`] = `
Object {
  "devMiddleware": Object {
    "logTime": true,
  },
  "hotClient": Object {
    "hmr": false,
    "logTime": true,
  },
  "logTime": true,
}
`;

exports[`flags compound logTime/logLevel 1`] = `
Object {
  "devMiddleware": Object {
    "logLevel": "warn",
    "logTime": true,
  },
  "hotClient": Object {
    "logLevel": "warn",
    "logTime": true,
  },
  "logLevel": "warn",
  "logTime": true,
}
`;
